src/main/java/com/contentgrid/gateway/security/jwt/issuer/JwtAuthenticationTokenAuthenticationInformationResolver.java

```diff
@@ -0,0 +1,27 @@
+package com.contentgrid.gateway.security.jwt.issuer;
+
+import java.time.Instant;
+import java.util.Optional;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import reactor.core.publisher.Mono;
+
+public class JwtAuthenticationTokenAuthenticationInformationResolver implements AuthenticationInformationResolver {
+
+    @Override
+    public Mono<AuthenticationInformation> resolve(Authentication authentication) {
+        if (authentication instanceof JwtAuthenticationToken jwt) {
+            var token = jwt.getToken();
+
+            return Mono.just(AuthenticationInformation.builder()
+                    .issuer(token.getIssuer().toString())
+                    .subject(token.getSubject())
+                    .expiration(Optional.ofNullable(token.getExpiresAt()).orElse(Instant.MAX))
```

Comment:
isn't the `orElse(Instant.MAX)` a bit dangerous ?

Reply:
This is the expiration of the existing token; not of the one that will be issued.
The token that is issued is capped at 5 minutes, or the expiration time of the existing token, whichever is smaller.